### PR TITLE
Opssight versioning

### PR DESCRIPTION
**Pull request by mphammer**

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/319#" title="Author self-approved">mphammer</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [mphammer]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/api/opssight/v1/types.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ under the License.
 package v1
 
 import (
-	"github.com/blackducksoftware/synopsys-operator/pkg/api/blackduck/v1"
+	v1 "github.com/blackducksoftware/synopsys-operator/pkg/api/blackduck/v1"
```
Comment (msenmurugan) [inline pkg/api/opssight/v1/types.go]: pls remove v1. I think, we need to compare our VSCode format

File: pkg/apps/blackduck/latest/creater.go [hunk 2]
```diff
@@ -238,7 +238,7 @@ func (hc *Creater) initPostgres(bdspec *blackduckapi.BlackduckSpec) error {
 
 	for dbInitTry := 0; dbInitTry < math.MaxInt32; dbInitTry++ {
 		// get the secret from the default operator namespace, then copy it into the hub namespace.
-		adminPassword, userPassword, postgresPassword, err = bdutils.GetDefaultPasswords(hc.KubeClient, hc.Config.Namespace)
+		adminPassword, userPassword, postgresPassword, _, err = util.GetDefaultPasswords(hc.KubeClient, hc.Config.Namespace)
```
Comment (msenmurugan) [inline pkg/apps/blackduck/latest/creater.go]: Probably you need to fix the merge conflict of this PR with Yash changes related to password

File: pkg/apps/blackduck/latest/deployer.go [hunk 3]
```diff
@@ -52,7 +51,7 @@ func (hc *Creater) getPostgresComponents(blackduck *blackduckapi.Blackduck) (*ap
 		adminPassword = blackduck.Spec.ExternalPostgres.PostgresAdminPassword
 		userPassword = blackduck.Spec.ExternalPostgres.PostgresUserPassword
 	} else {
-		adminPassword, userPassword, _, err = bdutil.GetDefaultPasswords(hc.KubeClient, hc.Config.Namespace)
+		adminPassword, userPassword, _, _, err = util.GetDefaultPasswords(hc.KubeClient, hc.Config.Namespace)
```
Comment (msenmurugan) [inline pkg/apps/blackduck/latest/deployer.go]: Probably you need to fix the merge conflict of this PR with Yash changes related to password

File: pkg/apps/opssight/opssight.go [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+
+	creater, err := o.getCreater(ops.Spec.Version) // get Creater for the OpsSight Version
+	if err != nil {
+		return fmt.Errorf("failed to get OpsSight creater for version %s: %s", ops.Spec.Version, err)
+	}
+	return creater.Ensure(ops) // Ensure the OpsSight
+}
+
+// Delete will delete the Black Duck OpsSight
+func (o OpsSight) Delete(namespace string) error {
```
Comment (jdartigalongue) [inline pkg/apps/opssight/opssight.go]: @msenmurugan @mphammer  Shouldn't this function delete Prometheus too?

Comment (msenmurugan) [reply pkg/apps/opssight/opssight.go]: Yup. It needs too.. It is already deleting... All the objects are under app=opssight @jdartigalongue

Comment (jdartigalongue) [reply pkg/apps/opssight/opssight.go]: @msenmurugan It currently deletes all the replication controllers with label 'app=opssight'  but prometheus is a deployment and not a RC.

**Comment by operator-r2d2** [comment]
@mphammer: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: pkg/apps/opssight/opssight.go (hunk 4 above)
Comment (mphammer) [reply pkg/apps/opssight/opssight.go]: @msenmurugan @jdartigalongue I made it so if there are Replication Controllers that do not belong to OpsSight in the namespace, then Delete() will delete all Replication Controllers **and** Deployments that have the label "app=opssight" from the namespace.

**Comment by operator-r2d2** [comment]
@mphammer: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by operator-r2d2** [comment]
@mphammer: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Pull request closed without merging**